src/app/dynamic-forms/service/cinchy-dropdown-dataset/cinchy-dropdown-dataset.service.ts

```diff
@@ -111,39 +111,85 @@ export class DropdownDatasetService {
 
     let optionArray: DropdownOption[] = [];
 
-    (await this._cinchyService.executeCsql(dataSetQuery, params).toPromise()).queryResult.toObjectArray().forEach(function (row) {
+    try {
+      (await this._cinchyService.executeCsql(dataSetQuery, params).toPromise()).queryResult.toObjectArray().forEach(function (row) {
 
-      // TODO For now only showing one display column (display-0)
-      const label = row["display-0"] ? `${row["Label"]}, ${row["display-0"]}` : row["Label"];
+        let label = row["Label"];
 
-      optionArray.push(new DropdownOption(row["Id"].toString(), row["Label"]?.toString(), label));
-    });
+        let displayIndex = 0;
 
-    const result = new DropdownDataset(optionArray);
+        do {
+          if (row[`display-${displayIndex}`]) {
+            label += `, ${row[`display-${displayIndex}`]}`;
+          }
+        }
+        while (++displayIndex);
```

Comment:
When does this while loop stop? Doesn't `++displayIndex` keep adding 1 to itself?

Reply:
Good catch, though I already fixed it